src/OpenTelemetry/Metrics/Aggregators/Aggregator.cs

```diff
@@ -47,5 +62,23 @@ public abstract class Aggregator<T>
         /// </summary>
         /// <returns><see cref="AggregationType"/>.</returns>
         public abstract AggregationType GetAggregationType();
+
+        /// <summary>
+        /// Get the last interval end time before <see cref="Checkpoint"/> was called.
+        /// </summary>
+        /// <returns>The end timestamp of the last aggregated checkpoint.</returns>
+        protected DateTimeOffset GetLastEndTimestamp()
+        {
+            return new DateTimeOffset(this.startTimeTicks, TimeSpan.Zero).Subtract(TimeSpan.FromMilliseconds(1));
```

Comment:
Since we're dealing in ticks elsewhere, should `TimeSpan.FromTicks` be used here instead of `TimeSpan.FromMilliseconds`?

Reply:
Just curious, why are we subtracting one millisecond from the time?

Reply:
Done

Reply:
@CodeBlanch Some cloud providers need to know the entire window for which gauge/measure metrics were collected. A single timestamp is not sufficient. What I'm doing here is tracking both the current aggregation windows start time and the previous windows start time. 

`Checkpoint()` is called and we save the current aggregation windows start time before initiating the new window. The new windows start time is saved now as the current. It's only some time shortly after that we will actually create the metric data object, which is when we need to set the timestamps. I am doing a subtraction here, because it is important that the windows are as close together as possible to represent reality but they absolutely cannot intersect. If they did, then some of the metric data would belong to both windows. So basically I'm tracking the start times and calculating the end timestamp of the last aggregated window to be "new window start time" - 1 time unit.

Reply:
Current implementation actually isn't true to the times the data was collected. `Checkpoint()` is called and then some time shortly after `ToMetricData()`, and the timestamp is getting set to that point in time. Technically, it's possible that some metrics may have already been recorded in the new aggregation window before `ToMetricData()` was called. And like I said, end timestamp isn't really enough information.